lms/djangoapps/student_account/views.py

```diff
@@ -302,6 +306,69 @@ def _external_auth_intercept(request, mode):
         return external_auth_register(request)
 
 
+def get_user_orders(user):
+    """Given a user, get the detail of all the orders from the Ecommerce service.
+
+    Arguments:
+        user (User): The user to authenticate as when requesting ecommerce.
+
+    Returns:
+        list of dict, representing orders returned by the Ecommerce service.
+    """
+    no_data = []
+    commerce_configuration = CommerceConfiguration.current()
+    if not commerce_configuration.enabled:
+        log.warning('%s configuration is disabled.', commerce_configuration.API_NAME)
+        return no_data
+
+    if commerce_configuration.is_cache_enabled:
+        cache_key = commerce_configuration.CACHE_KEY + '.' + user.username
+        cached = cache.get(cache_key)
+        if cached:
+            return cached
+
+    user_orders = []
+    allowed_course_modes = ['professional', 'verified', 'credit']
+    try:
+        response = ecommerce_api_client(user).orders.get()
+        commerce_user_orders = response.get('results', no_data)
+        page = 1
+        next_page = response.get('next', None)
+        while next_page:
```

Comment:
In particular, this code to traverse a paginated API can be re-used from https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L73.